source/docs/hardware-reference/talonfx/improving-performance-with-current-limits.rst

```diff
@@ -0,0 +1,217 @@
+
+Supply current is the current drawn from the battery. As a result, limiting supply current can be useful to prevent fuses from resetting and breakers from tripping, as well as to improve the longevity of the battery.
+
+.. tip:: Since stator current limits also limit supply current, it is often not necessary to enable both limits.
+
+When the ``SupplyCurrentThreshold`` has elapsed for ``SupplyTimeThreshold`` amount of time, the supply current limiter will activate and reduce motor output until supply current is within range of the limit.
+
+In the rare case where the robot experiences brownouts despite configuring stator current limits, a supply current limit can also further help avoid brownouts. However, such brownouts are most commonly caused by a bad battery or poor power wiring, so those should be examined first.
+
+How to Budget Limits
+--------------------
+
+When budgeting current limits for FRC, the most robust strategy is to run a robot in match-like conditions and observe current draw through the course of the robot operation. One thing to be aware of when reading the following section is that the robot will not be under peak draw of **all** mechanisms at the same time.
+
+Improving Battery Longevity
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While supply limits can be estimated using battery datasheets and average mechanism current draw, the easiest method is to graph and reduce. By applying a conservative supply limit to mechanisms that can operate with minimal current draw, you can improve the performance of other mechanisms (e.g. capping your intake supply to increase the amount of current a swerve drivetrain can draw).
+
+1. Plot a mechanism supply currents throughout the match
+2. Reduce your supply current until your performance begins to suffer (unable to intake items, etc). Increase this value slightly for some comfort room.
+3. Repeat this for all mechanisms on a fresh battery each time. Time how long it takes before brownouts are regular and battery capacity has dropped.
+
+For example, a user may have the following mechanisms and supply limits.
+
+- x4 Kraken(s) on swerve drive - 60 A supply
+- x4 Kraken(s) on swerve azimuth - 20 A supply
+- x1 Kraken(s) on elevator - 30 A supply
+- x1 Kraken(s) on intake - 15 A supply
+
+This would yield peak supply current of ~365 A for a worst case scenario. This draw is extremely unlikely as peak supply current is often extremely brief (for example, 60 A on all 4 swerve drive motors will likely be for less than 2 seconds), and all mechanisms will not be under peak load at the same time. A more common scenario is 4 swerve drive motors accelerating at the same time for a peak supply current of 240 A.
+
+.. math::
+
+   (60 * 4) + (20 * 4) + (30 * 1) + (15 * 1) \approx 365\text{ A}
+
+Reduce your limits until your battery life is in an acceptable range.
+
+Preventing Brownouts
+^^^^^^^^^^^^^^^^^^^^
```

Comment:
This section needs to be adjusted based on the "Stator and Supply Current Limits" section. In particular, I'm not sure the calculator makes sense now (nor do I see how to make it make sense).

We might want to make this section based more on experimental testing instead of theoretical because of stator limits. Theoretical only really works with supply limits.